cpp/src/join/join_compute_api.h

```diff
@@ -239,7 +211,7 @@ gdf_error compute_hash_join(
 #ifdef HT_LEGACY_ALLOCATOR
   using multimap_type = concurrent_unordered_multimap<hash_value_type,
                                                       output_index_type,
-                                                      size_type,
+                                                      size_t,
```

Comment:
In this case, `size_t` is appropriate since it's the type that is used internal to the map class. A column may be limited to `2^31-1`, but the hash table may need to be larger than that (currently, the hash table is always 2x the size of the input column).